### ISSUE TITLE
[a11y][Android] Checkbox and Switch state changes are not announced on Android 16 (SDK 36)

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#181321]
### Steps to reproduce

1. Build an app that features checkboxes and/or switches
2. Run it on a device running Android 16 (SDK 36)
3. Enable TalkBack
4. Observe the behavior of toggling checkboxes and switches

### Expected results

Checkboxes should announce "checked" or "unchecked" when toggled. Switches should announce "on" or "off" when toggled.

### Actual results

Nothing is announced. On Android 15 (SDK 35) and older versions, announcements work correctly.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  bool _checked = false;
  bool _toggled = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: ListView(
        children: [
          Text('Checkbox is ${_checked ? 'checked' : 'unchecked'}.'),
          CheckboxListTile(
            title: Text('Checkbox'),
            value: _checked,
            onChanged: (value) {
              if (value == null) return;
              setState(() {
                _checked = value;
              });
            },
          ),
          Text('Switch is ${_toggled ? 'toggled' : 'untoggled'}.'),
          SwitchListTile(
            title: Text('Switch'),
            value: _toggled,
            onChanged: (value) {
              setState(() {
                _toggled = value;
              });
            },
          ),
        ],
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

* Video on an older version of Android: https://github.com/user-attachments/assets/3c77478e-dd53-4b7f-8633-2a7cebc2bed9
* Video on Android 36: https://github.com/user-attachments/assets/437893c4-f9b2-4ecd-a6bd-2a10c22ff122

</details>

### Logs

N/A

### Flutter Doctor output

N/A, google internal client

**Comment by bourdakos1** [comment]
@hannah-hyj I'm not sure if its a regression, QA caught the issue in the latest a11y testing pass. It's reproducible on both Windows and macOS desktop platforms (not web).

**Comment by hannah-hyj** [comment]
I did more tests on Android. Originally I only tested it on emulator, on emulator with Android 16 it works fine.
But I can reproduce it with an Android device, I need to look into it more.
